lib/db/mem.js

```diff
@@ -474,15 +463,14 @@ module.exports = function (log, error) {
         function(account) {
           return Object.keys(account.devices)
             .map(
-              function (id) {
-                var device = account.devices[id]
+              function (deviceKey) {
+                var device = Object.assign({}, account.devices[deviceKey])
```

Comment:
@shane-tomlinson showed me this sweet trick the other day, [object spreading](https://github.com/tc39/proposal-object-rest-spread#spread-properties):

```js
var device = { ...account.devices[deviceKey] }
```

Not that we have to use it of course, but it's my brand new hammer and all I see are nails.

Reply:
I like this, but unfortunately `grunt eslint` seems to choke on it, and I don't have the bandwidth to debug right now so I'm going to revert.